src/DynamoCore/Logging/DynamoAnalyticsClient.cs

```diff
@@ -357,26 +358,52 @@ public Task<IDisposable> CreateTaskTimedEvent(Categories category, string variab
 
         [Obsolete("Property will become private in Dynamo 4.0, please use CreateTaskCommandEvent")]
         public IDisposable CreateCommandEvent(string name, string description, int? value)
+        {
+            return CreateCommandEvent(name, description, value, null);
+        }
+
+        private IDisposable CreateCommandEvent(string name, string description, int? value, Dictionary<string, object> parameters = null)
         {
             serviceInitialized.Wait();
 
             lock (trackEventLockObj)
             {
                 if (!ReportingAnalytics) return Disposable;
 
-                var e = new CommandEvent(name) { Description = description, Value = value };
+                var e = new CommandEvent(name) { Description = description };
+
+                if (value != null)
+                {
+                    e.Value = value;
+                }
+                
+                if (parameters != null)
+                {
+                    foreach (var item in parameters)
+                    {
+                        e[item.Key] = item.Value;
+                    }
+                }    
+
                 e.Track();
                 return e;
             }
         }
 
-        public Task<IDisposable> CreateTaskCommandEvent(string name, string description, int? value)
+        public Task<IDisposable> CreateTaskCommandEvent(string name, string description, int value)
```

Comment:
I am likely missing something but - why doesn't one of these methods exist that expects both a value and attributes dictionary?

Reply:
Argggg, part of the interface...

Reply:
It's okay to break interfaces now though - right?

Reply:
we broke it 3 weeks ago 
I hope it is ok now too